dashboard/src/entities/weeklyHeatmap.ts

```diff
@@ -0,0 +1,31 @@
+import { z } from 'zod';
+
+export const HeatmapMetricSchema = z.enum([
+  'pageviews',
+  'unique_visitors',
+  'sessions',
+  'bounce_rate',
+  'pages_per_session',
+  'session_duration',
+]);
+
+export type HeatmapMetric = z.infer<typeof HeatmapMetricSchema>;
+
+export const WeeklyHeatmapRowSchema = z.object({
+  // ISO date truncated to hour for the bucket start (mainly for traceability)
+  date: z.string(),
+  // 1-7 where 1 = Monday, 7 = Sunday (ClickHouse toDayOfWeek ISO mapping)
+  weekday: z.number().int().min(1).max(7),
+  // 0-23 hour of day
```

Comment:
I think using `toDayOfWeek` is what causes the incorrect date mapping. It should be done the same way as the rest of the application handles the date mapping between client / server / clickhouse.